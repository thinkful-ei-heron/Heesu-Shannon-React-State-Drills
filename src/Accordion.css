ul {
  background-color: grey;
  width: 50%;
  margin-left:auto;
  margin-right: auto;
  list-style:none;
  padding: 0;
}

li {
  text-align: center;
}

button:hover {
  background-color: lightblue;
}

p{
  width: 90%;
  border: 1px solid black;
  margin-left:auto;
  margin-right:auto;
}