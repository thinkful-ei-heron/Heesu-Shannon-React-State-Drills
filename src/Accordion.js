import React from 'react';

class Accordian extends React.Component {
    static defaultProps = {
        sections: []
    };
    state = {
        displayToggle: null,
    }

    handleClick = (i) => { 
            this.setState({displayToggle: i})
    }

    render() {
        
        return(
            <ul>
                {(this.props.sections !== []) ? 
                (this.props.sections.map((item, i) => {
                    return (
                        <li key= {i}>
                            <button onClick={()=> this.handleClick(i)}>
                                {item.title}
                            </button>
                            {(this.state.displayToggle === i) && <p>{item.content}</p>}
                    </li> )  
                })) : ''
            }
            </ul>
        )
    }

}



export default Accordian;