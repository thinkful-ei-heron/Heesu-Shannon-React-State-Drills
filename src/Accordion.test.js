import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Accordion from './Accordion';
import { configure } from 'enzyme';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});


const sections = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
];




it('renders correctly', () => {
  const tree = renderer
    .create(<Accordion sections={sections} 
      />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders empty when no section supplied', () => {
  const tree = renderer
    .create(<Accordion 
      />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('opens a clicked section', () => {
  const wrapper = shallow(<Accordion sections={sections} />)
  wrapper.find('button').at(1).simulate('click')
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('opens only one section when multiple are clicked', () => {
  const wrapper = shallow(<Accordion sections={sections} />)
  wrapper.find('button').at(1).simulate('click')
  wrapper.find('button').at(2).simulate('click')
  expect(toJson(wrapper)).toMatchSnapshot()
})