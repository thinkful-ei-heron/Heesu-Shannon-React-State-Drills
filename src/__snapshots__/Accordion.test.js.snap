// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`opens a clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`opens only one section when multiple are clicked 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`renders correctly 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`renders empty when no section supplied 1`] = `<ul />`;
